File: apps/DriftDate/android/core/src/main/java/com/americangroupllc/drift/core/models/Enums.kt
package com.americangroupllc.drift.core.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class Layer {
    @SerialName("zip")    ZIP,
    @SerialName("county") COUNTY,
    @SerialName("state")  STATE,
    @SerialName("server") SERVER;
}

@Serializable
enum class Intent {
    @SerialName("dating")     DATING,
    @SerialName("serious")    SERIOUS,
    @SerialName("friendship") FRIENDSHIP,
    @SerialName("open")       OPEN;
}

@Serializable
enum class Tone {
    @SerialName("slow")          SLOW,
    @SerialName("energetic")     ENERGETIC,
    @SerialName("deep")          DEEP,
    @SerialName("meetup_ready")  MEETUP_READY;
}

@Serializable
enum class WaveStatus {
    @SerialName("pending") PENDING,
    @SerialName("matched") MATCHED,
    @SerialName("passed")  PASSED;
}

File: apps/DriftDate/shared/DriftCore/Sources/DriftCore/Models/Enums.swift
import Foundation

/// Discovery layers, in increasing radius.
public enum Layer: String, Codable, CaseIterable, Sendable {
    case zip
    case county
    case state
    case server
}

public enum Intent: String, Codable, CaseIterable, Sendable {
    case dating
    case serious
    case friendship
    case open
}

public enum Tone: String, Codable, CaseIterable, Sendable {
    case slow
    case energetic
    case deep
    case meetupReady = "meetup_ready"
}

public enum WaveStatus: String, Codable, Sendable {
    case pending
    case matched
    case passed
}
